pkg/operator/rules.go

```diff
@@ -126,9 +127,80 @@ func (r *rulesReconciler) Reconcile(ctx context.Context, req reconcile.Request)
 	if err := r.ensureRuleConfigs(ctx, projectID, location, cluster); err != nil {
 		return reconcile.Result{}, fmt.Errorf("ensure rule configmaps: %w", err)
 	}
+
+	if err := r.scaleRuleConsumers(ctx); err != nil {
+		return reconcile.Result{}, fmt.Errorf("scale rule consumers: %w", err)
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *rulesReconciler) scaleRuleConsumers(ctx context.Context) error {
+	var desiredReplicas int32
+
+	var hasAnyRules bool
+	for _, check := range []RuleCheck{r.hasRules, r.hasClusterRules, r.hasGlobalRules} {
+		hasRules, err := check(ctx)
+		if err != nil {
+			return err
+		}
+		if hasRules {
+			hasAnyRules = true
+			break
+		}
+	}
+	if hasAnyRules {
+		desiredReplicas = 1
+	}
+
+	var alertManagerStatefulSet appsv1.StatefulSet
+	if err := r.client.Get(ctx, client.ObjectKey{Namespace: r.opts.OperatorNamespace, Name: "alertmanager"}, &alertManagerStatefulSet); err != nil {
+		return err
```

Comment:
What if the user deleted alertmanager but not rule-evaluator. In that case this code will just return err and not scale the rule-evaluator. Let's just to continue in that case? This is a suggestion of what that code could look like:

```suggestion
	var alertManagerStatefulSet appsv1.StatefulSet
	err := r.client.Get(ctx, client.ObjectKey{Namespace: r.opts.OperatorNamespace, Name: NameAlertmanager}, &alertManagerStatefulSet)
	if err != nil {
		// only return error if alertmanager exists
		if !apierrors.IsNotFound(err) { 
			return err
		}
	} else {
		if *alertManagerStatefulSet.Spec.Replicas != desiredReplicas {
			*alertManagerStatefulSet.Spec.Replicas = desiredReplicas
			if err := r.client.Update(ctx, &alertManagerStatefulSet); err != nil {
				return err
			}
		}
	}
```